### PR TITLE
Allow showing a template or static string in a region

### PR DESCRIPTION
**Pull request by paulfalgout**
Resolves https://github.com/marionettejs/backbone.marionette/issues/2317 and https://github.com/marionettejs/backbone.marionette/issues/2290

I think this feature will be better when we can create regions on the fly from selectors.  For instance imagine this API:

```js
myApp.showView(`
  <div class="header">Hello ${ user.get('name') }</div>
  <div class="main-region main"></div>
  <div class="sidebar-region sidebar"></div>
`);

myApp.getView().showChildView('.main-region', new MainView({ model }));

myApp.getView().showChildView('.sidebar-region', `
  <h2>User Info</h2>
  <div>Select an option to see more info...</div>
`);
```

### CODE REVIEW

**Comment by paulfalgout** [comment]
So initially I made it such that you could pass `show(template, viewOptions)`

This was awkward for two reasons.  
1. `viewOptions` could contain `template` in which case the 1st argument is useless.
2. The 2nd argument is also used for show options.  I initially thought these would be removed in the future, but `replaceElement` is likely to stay a show option.

So I've added a commit changing this to:
```js
myRegion.show({
  template: _.template('Hello <%= place %>!'),
  model: new Backbone.Model({ place: 'World' })
});
```

File: docs/marionette.region.md [hunk 1]
```diff
@@ -261,6 +262,26 @@ mainRegion.hasView() // true
...
 [remove the existing View](#emptying-a-region) before showing the new one.
 
+## Showing a Template
+
+You can show a template or a string directly into a region. Additionally you can pass an object literal containing a template and any other view options. Under the hood a `Marionette.View` is instantiated using the template.
+
+```javascript
+var myView = new MyView();
+
+var template = _.template('This is the <%= section %> page');
```
Comment (scott-w) [inline docs/marionette.region.md]: I'd prefer to use `<%- %>` here for documentation purposes to promote escaping where possible.

Comment (paulfalgout) [inline docs/marionette.region.md]: oh good catch.. I reformat my underscore template to use `{{}}` and `{{{}}}` so I don't have any clue what the normal is :-D

**Comment by paulfalgout** [comment]
I updated the docs line for @scott-w and updated the same issue in the specs.. so I'm going to go ahead and get this merged in.

**Pull request merged**